spec/jobs/allocation_job_spec.rb

```diff
@@ -279,33 +338,102 @@
+          create(:allocation_batch_job, urn: school1.urn, allocation_delta: '0', order_state: 'cannot_order')
+        end
+        let(:batch_allocation_job2) do
+          create(:allocation_batch_job, urn: school2.urn, allocation_delta: '0', order_state: 'cannot_order')
+        end
+
+        it 'does not change the allocation' do
+          expect {
+            described_class.perform_now(batch_allocation_job1)
+          }.to change { school1.std_device_allocation.reload.allocation }.by(0)
+
+          expect {
+            described_class.perform_now(batch_allocation_job2)
+          }.to change { school2.std_device_allocation.reload.allocation }.by(0)
+        end
+
+        it 'does not change the raw_allocation' do
+          expect {
+            described_class.perform_now(batch_allocation_job1)
+          }.to change { school1.std_device_allocation.reload.raw_allocation }.by(0)
+
+          expect {
+            described_class.perform_now(batch_allocation_job2)
+          }.to change { school2.std_device_allocation.reload.raw_allocation }.by(0)
+        end
+
+        it 'updates the raw_cap to match raw_devices_ordered' do
+          described_class.perform_now(batch_allocation_job1)
+          described_class.perform_now(batch_allocation_job2)
+
+          expect(school1.std_device_allocation.raw_cap).to eql(school1.std_device_allocation.raw_devices_ordered)
+          expect(school2.std_device_allocation.raw_cap).to eql(school2.std_device_allocation.raw_devices_ordered)
+        end
+
+        it 'returns false for devices_available_to_order?' do
+          described_class.perform_now(batch_allocation_job1)
+          described_class.perform_now(batch_allocation_job2)
+
+          expect(school1.std_device_allocation.reload.devices_available_to_order?).to be(false)
+          expect(school2.std_device_allocation.reload.devices_available_to_order?).to be(false)
+        end
+      end
+
       context 'reducing allocation' do
-        let(:batch_job) { create(:allocation_batch_job, urn: school1.urn, allocation_delta: '-1', order_state: 'can_order') }
+        let(:batch_deallocation_job) { create(:allocation_batch_job, urn: school1.urn, allocation_delta: '-1', order_state: 'can_order') }
 
         it 'reduces allocation' do
           expect {
-            described_class.perform_now(batch_job)
+            described_class.perform_now(batch_deallocation_job)
           }.to change { school1.std_device_allocation.reload.raw_allocation }.by(-1)
         end
       end
 
-      context 'maintain part of allocation if already ordered' do
-        let!(:school) { create(:school, :with_std_device_allocation_partially_ordered) }
-        let(:batch_job) { create(:allocation_batch_job, urn: school.urn, allocation_delta: '-100', order_state: 'cannot_order') }
+      context 'reduce and maintain part of allocation if already ordered' do
+        let!(:devices_available_to_deallocate) { [school1_devices_available_to_order, 198].min }
```

Comment:
Don't worry about it now but that makes me think too much

Reply:
Agreed. We already know there is refactoring work to be done here so we will clear this up in the subsequent PR.